pkgs/build-support/fetchpypilegacy/default.nix

```diff
@@ -0,0 +1,44 @@
+# Fetch from PyPi legacy API as documented in https://warehouse.pypa.io/api-reference/legacy.html
+{ runCommand
+, lib
+, python3
+}:
+{
+  # package name
+  pname,
+  # Package index
+  url ? null,
```

Comment:
This is redundant since there's already a `urls` argument.

Reply:
It's more ergonomic to support either a singular source or a list.
Commonly you will probably only have one source to consider, but sometimes you have more.

This is analogous to how fetchurl supports either URL or urls.